docs/plat/rpi3.rst

```diff
@@ -0,0 +1,297 @@
+``0x00000000`` to ``0x3F000000``, it just has different names to simulate a real
+secure platform!
+
+.. code::
+
+    0x00000000 +-----------------+
+               |       ROM       | BL1
+    0x00010000 +-----------------+
+               |       FIP       |
+    0x00200000 +-----------------+
+               |                 |
+               |       ...       |
+               |                 |
+    0x01000000 +-----------------+
+               |     Kernel      |
+               +-----------------+
+               |                 |
+               |       ...       |
+               |                 |
+    0x02000000 +-----------------+
+               |       DTB       |
+               +-----------------+
+               |                 |
+               |       ...       |
+               |                 |
+    0x10000000 +-----------------+
+               |   Secure SRAM   | BL2, BL31
+    0x10100000 +-----------------+
+               |   Secure DRAM   |
+    0x10300000 +-----------------+
+               | Non-secure DRAM | BL33
+    0x11000000 +-----------------+
+               |                 |
+               |       ...       |
+               |                 |
+    0x3F000000 +-----------------+
+               |       I/O       |
+    0x40000000 +-----------------+
+
+The area between ``0x10000000`` and ``0x11000000`` has to be protected so that
+the kernel doesn't use it. That is done by adding ``memmap=256M$16M`` to the
+command line passed to the kernel. See the `Setup SD card`_ instructions to see
+how to do it.
+
+The last 16 MiB of DRAM can only be accessed by the VideoCore, that has
+different mappings than the Arm cores, where the I/O addresses don't overlap the
+DRAM. The memory used by the VideoCore is always placed at the end of the DRAM,
+so this space isn't wasted.
+
+Considering the 128 MiB allocated to the GPU and the 16 MiB allocated for the
+Trusted Firmware, there are 880 MiB available for the kernel.
+
+Boot sequence
+~~~~~~~~~~~~~
+
+The boot sequence of the Trusted Firmware is the usual one. The only difference
+is BL33, which is booted in AArch32 Hypervisor mode so that it can jump to the
+kernel in the same mode and let it take over that privilege level. If BL33 was
+running in EL2 in AArch64, it could only jump to the kernel in AArch32 in
+Supervisor mode.
+
+Jumping to the kernel
+~~~~~~~~~~~~~~~~~~~~~
+
+The `Linux kernel tree`_ has instructions on how to jump to the Linux kernel
+correctly in the file ``Documentation/arm/Booting``. In short:
+
+.. code::
+
+    CPU register settings:
+      r0 = 0
+      r1 = Machine type number (3138 for the Rasbperry Pi 3)
+      r2 = Physical address of the device tree block (dtb) in RAM
+
+    CPU mode:
+      All forms of interrupts must be disabled (IRQs and FIQs)
+
+The AArch32 bootstrap takes care of the setup of the registers, and BL2 setups
+the BL33 context so that all interrupts are disabled.
```

Comment:
This is specific to the AArch32 kernel. The ARM64 kernel has a different boot interface.

Reply:
I've removed this part and just kept the reference to the file with the information.